src/conn/conn_sweep.c

```diff
@@ -183,7 +192,7 @@ __sweep_remove_one(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
     WT_DECL_RET;
 
     /* Try to get exclusive access. */
-    WT_RET(__wt_try_writelock(session, &dhandle->rwlock));
+    WT_RET(__wt_session_dhandle_try_writelock(session));
 
     /*
      * If there are no longer any references to the handle in any sessions, attempt to discard it.
```

Comment:
@agorrod Why don't we release the lock in case we can't discard the dhandle? Do you think we should unlock it before returning `EBUSY`?

Reply:
Oh, my bad didn't notice that it's `WT_ERR`. All good.